### PR TITLE
DEV-3300 | Expose available Mailchimp email lists in the revenue program serializer

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

This PR should not be merged until DEV-3297 has been merged (as it presupposes some changes from that PR).

Additionally, although this PR is code complete, there is no way to test it yet (either locally or in review app), as we don't have SPA-side UI components to support the OAuth flow, and in order to determine if the code changes in here actually work, we need to be able to complete the Oauth flow.

#### What's this PR do?

This PR adds a new property — `RevenueProgram.mailchimp_email_lists`— and exposes that property on the RP serializer as `RevenueProgramSerializer.mailchimp_email_lists`.

Note that this PR pulls in some changes from DEV-3297 that have not yet been merged into main:

- apps/organizations/migrations/0016_DEV-3297_add_mailchimp_fields.py
- Some changes in apps.organizations.models.py that updated the model (adding fields in the migration above).
- Some changes in revengine/settings/base.py 

Additionally, this PR:

- Adds `mailchimp-marketing` as a dependency
- Updates Django to 3.2.18 because there was a safety warning

#### Why are we doing this? How does it help us?

This will allow org users to select which mailchimp list to send events to.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Cannot be manually tested at this point. I'll circle back and fill in these instructions once we have UI support for Oauth flow and I can go through the flow myself.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-3300](https://news-revenue-hub.atlassian.net/browse/DEV-3300)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

[DEV-3300]: https://news-revenue-hub.atlassian.net/browse/DEV-3300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/organizations/tests/test_models.py [hunk 1]
```diff
@@ -143,6 +144,41 @@ def test_apple_pay_domain_verification_when_stripe_error(self, mocker):
...
+        revenue_program.mailchimp_server_prefix = None
+        revenue_program.mailchimp_access_token = "123456"
+        revenue_program.save()
+        assert revenue_program.mailchimp_email_lists == []
+
+    def test_mailchimp_email_lists_property_when_missing_access_token(self, revenue_program):
+        revenue_program.mailchimp_server_prefix = "us1"
+        revenue_program.mailchimp_access_token = None
+        revenue_program.save()
+        assert revenue_program.mailchimp_email_lists == []
```
Comment (njharman) [inline apps/organizations/tests/test_models.py:166]: [me talking myself into this is non-issue, will resolve]
Shouldn't these also mock get_all_lists to return value?  ... I guess it's minor/moot, We know the guard clause comes before list request. It's unlikely future mailchip_email_lists would refactor to cut/break guard and just rely try/catch to return empty list... [just triggered my "Review Mode pattern recognition...]

could assert no logging, to ensure setting checked and empty list not just from failed api call

File: revengine/settings/base.py [hunk 2]
```diff
@@ -548,3 +548,7 @@
 # webpack) instead of getting sent to localhost:8000, which is what would happen by default in local
 # dev environment.
 STRIPE_ACCOUNT_LINK_RETURN_BASE_URL = os.getenv("STRIPE_ACCOUNT_LINK_RETURN_BASE_URL", None)
+
+# These `MAILCHIMP_` values are used by code that makes requests to mailchimp on behalf of org users
+MAILCHIMP_CLIENT_ID = os.getenv("MAILCHIMP_CLIENT_ID", None)
+MAILCHIMP_CLIENT_SECRET = os.getenv("MAILCHIMP_CLIENT_SECRET", None)
```
Comment (njharman) [inline revengine/settings/base.py:554]: These don't seem to be used anywhere. Are they used by mailchimp_marketing lib? comment should mention that.

**Review by njharman — APPROVED** [review]
Looks good, approved. Only one request to add to comment. I don't feel need to rereview for that.

**Comment by benjaminEwhite** [comment]
Closing and reopening after pushing upstream migration changes.

File: revengine/settings/base.py (hunk 2 above)
Comment (benjaminEwhite) [reply revengine/settings/base.py]: Yeah these actually aren't required here. I pulled them in from 3298 at beginning cause thought I would need them, and turns out I didn't, but didn't remember to remove.

523d362aaeae07

File: pyproject.toml [hunk 3]
```diff
@@ -73,6 +73,7 @@ uwsgi = "^2"
 whitenoise = "^6.2.0"
 google-cloud-pubsub = "^2.13"
 pygments = "^2.13.0"
+mailchimp-marketing = "^3.0.80"
```
Comment (x110dc) [inline pyproject.toml:76]: Prefer that this be `^3.0`

**Review by x110dc — CHANGES_REQUESTED** [review]
One super-small change request.

File: pyproject.toml (hunk 3 above)
Comment (benjaminEwhite) [reply pyproject.toml]: 95145589a90a27

**Review by x110dc — APPROVED** [review]

**Pull request merged**